tests/pages/IdentityOverviewPage.test.tsx

```diff
@@ -42,409 +15,444 @@ jest.mock("@src/dbClient", () => {
+          userId={userId}
+        />,
+      );
 
-      expect(keepEmailPrimaryButton.text()).toEqual("Cancel");
-      keepEmailPrimaryButton.simulate("click");
+      expect(
+        screen.getByRole("link", { name: "Proceed to validation" }),
+      ).toBeInTheDocument();
 
-      const confirmationBox = component
-        .find(ConfirmationBox)
-        .find({ open: true })
-        .hostNodes();
+      expect(
+        screen.getByRole("button", { name: /Delete this phone number/i }),
+      ).toBeInTheDocument();
 
-      expect(confirmationBox).toHaveLength(0);
+      expect(
+        screen.queryByRole("button", {
+          name: makeAsPrimaryRegExFactory(unvalidatedPhoneIdentity),
+        }),
+      ).not.toBeInTheDocument();
     });
 
-    test("we should be able to close the confirmation box by clicking on the ClickAwayListener", () => {
-      const component = mount(
-        <AccountApp>
-          <IdentityOverviewPage identity={nonPrimaryIdentity} userId={userId} />
-        </AccountApp>,
+    it("should display the awaiting validation badge and not primary one for an unvalidated phone number", () => {
+      render(
+        <IdentityOverviewPage
+          identity={unvalidatedPhoneIdentity}
+          userId={userId}
+        />,
       );
 
-      const makePrimaryButton = component
-        .find(Button)
-        .find({ variant: ButtonVariant.SECONDARY })
-        .at(0);
-      makePrimaryButton.simulate("click");
+      expect(screen.getByText("Awaiting validation")).toBeInTheDocument();
+      expect(screen.queryByText("Primary")).not.toBeInTheDocument();
+    });
+  });
+});
 
-      const clickAwayListener = component.find(ClickAwayListener);
-      expect(clickAwayListener).toHaveLength(1);
-      clickAwayListener.simulate("click");
+describe("ConfirmationBox", () => {
+  it("shouldn't display any confirmation box on first render", () => {
+    render(
+      <IdentityOverviewPage
+        identity={nonPrimaryEmailIdentity}
+        userId={userId}
+      />,
+    );
 
-      const confirmationBox = component
-        .find(ConfirmationBox)
-        .find({ open: true })
-        .hostNodes();
+    expect(
+      screen.queryByText(
+        `You are about to set ${nonPrimaryEmailIdentity.value} as primary.`,
+      ),
+    ).not.toBeInTheDocument();
 
-      expect(confirmationBox).toHaveLength(0);
-    });
+    expect(
+      screen.queryByText(
+        `You are about to delete ${nonPrimaryEmailIdentity.value}`,
+      ),
+    ).not.toBeInTheDocument();
+  });
 
-    test("the delete confirmation box shoud appear on click on 'delete this { type }' button", () => {
-      const component = mount(
-        <AccountApp>
-          <IdentityOverviewPage identity={nonPrimaryIdentity} userId={userId} />
-        </AccountApp>,
-      );
+  it("should display primary confirmation box after cliking on mark as primary button & close it by clicking on cancel button", async () => {
```

Comment:
```suggestion
  it("should display primary confirmation box after clicking on mark as primary button & close it by clicking on cancel button", async () => {
```

There are other places to change it too.